personio-recruiting-api.yaml

```diff
@@ -8,6 +8,7 @@ paths:
     parameters:
       - $ref: "#/components/parameters/XPersonioPartnerID"
       - $ref: "#/components/parameters/XPersonioAppID"
+      - $ref: "#/components/parameters/XCompanyID"
```

Comment:
Do we really want to expose that header? Just checking ...

Reply:
Yes, we need this header to be exposed for our service. It's already in the documentation, just moved it next to the other headers for consistency

Reply:
I see .. but I just want to double confirm with you how dangerous that is. Because now, you are allowing an external entity/consumer to pass the company id ... And there is a company-id = 0 that some systems interpret as all companies ... I think it's fine if you use it for internal communication right now, but I think this has to be omitted by the public documentation. And I believe that is not used anyways, since the monolith is taking care of injecting the company_id for upstream calls ...

Reply:
In this case these requests don't go through the monolith. They go directly to our [submit-application-api](https://gitlab.personio-internal.de/personio/recruiting/submit-application-api) service, and that's why we need the client to set this header for the multitenancy library to work. We're also validating in the app that this value is actually an existing company.

The old recruiting API did go through the monolith, but that one is now [deprecated](https://developer.personio.de/reference#deprecated-applicant) in favour of this one.